src/Arcus.Observability.Correlation/IServiceCollectionExtensions.cs

```diff
@@ -0,0 +1,70 @@
+using GuardNet;
+
+// ReSharper disable once CheckNamespace
+namespace Microsoft.Extensions.DependencyInjection
+{
+    /// <summary>
+    /// Extensions to set correlation access information on the <see cref="IServiceCollection"/>.
+    /// </summary>
+    public static class IServiceCollectionExtensions
+    {
+        /// <summary>
+        /// Adds operation and transaction correlation to the application.
+        /// </summary>
+        /// <param name="services">The services collection containing the dependency injection services.</param>
+        /// <param name="configureOptions">The function to configure additional options how the correlation works.</param>
+        public static IServiceCollection AddCorrelation(
+            this IServiceCollection services,
+            Action<CorrelationInfoOptions> configureOptions = null)
+        {
+            Guard.NotNull(services, nameof(services));
+
+            return AddCorrelation(services, new DefaultCorrelationInfoAccessor(), configureOptions);
+        }
+
+        /// <summary>
+        /// Adds operation and transaction correlation to the application.
+        /// </summary>
+        /// <param name="services">The services collection containing the dependency injection services.</param>
+        /// <param name="customCorrelationAccessor">The custom <see cref="ICorrelationInfoAccessor"/> implementation to retrieve the <see cref="CorrelationInfo"/>.</param>
+        /// <param name="configureOptions">The function to configure additional options how the correlation works.</param>
+        public static IServiceCollection AddCorrelation<TAccessor>(
+            this IServiceCollection services,
+            TAccessor customCorrelationAccessor,
+            Action<CorrelationInfoOptions> configureOptions = null)
+            where TAccessor : class, ICorrelationInfoAccessor
+        {
+            Guard.NotNull(services, nameof(services));
+            Guard.NotNull(customCorrelationAccessor, nameof(customCorrelationAccessor));
+
+            return AddCorrelation(services, serviceProvider => customCorrelationAccessor, configureOptions);
+        }
+
+        /// <summary>
+        /// Adds operation and transaction correlation to the application.
+        /// </summary>
+        /// <param name="services">The services collection containing the dependency injection services.</param>
+        /// <param name="createCustomCorrelationAccessor">The custom <see cref="ICorrelationInfoAccessor"/> implementation factory to retrieve the <see cref="CorrelationInfo"/>.</param>
+        /// <param name="configureOptions">The function to configure additional options how the correlation works.</param>
+        public static IServiceCollection AddCorrelation<TAccessor>(
+            this IServiceCollection services, 
+            Func<IServiceProvider, TAccessor> createCustomCorrelationAccessor,
+            Action<CorrelationInfoOptions> configureOptions = null)
```

Comment:
Should we also provide a generic type for the options so specific application correlation configurations can be configured here as well?

Reply:
Or... is the options here moved also too application specific.

Reply:
I'm a bit of expirimenting with maybe even make the `CorrelationInfo` type generic, so we can use directly (for example) the `MessageCorrelationInfo` instead of casting; when in a messaging context.

Reply:
Added generic `TOptions` for this.